File: editor/en/[2441]Largest Positive Integer That Exists With Its Negative.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int findMaxK(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for (int num : nums) {
            set.add(num);
        }

        int max = -1;

        for (int num : nums) {
            if (set.contains(-num) && num > max) {
                max = num;
            }
        }

        return max;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2441]Largest Positive Integer That Exists With Its Negative.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxK = function (nums) {
    const numSet = new Set(nums);
    let max = -1;

    for (const num of nums) {
        if (numSet.has(-num) && num > max) {
            max = num;
        }
    }

    return max;
};
//leetcode submit region end(Prohibit modification and deletion)
